test/Confluent.Kafka.IntegrationTests/Tests/Producer_CustomPartitioner.cs

```diff
@@ -0,0 +1,140 @@
+                    {
+                        Assert.StartsWith($"test key ", dr.Message.Key);
+                        Assert.StartsWith($"test val ", dr.Message.Value);
+                        var expectedPartition = int.Parse(dr.Message.Key.Split(" ").Last());
+                        Assert.Equal(ErrorCode.NoError, dr.Error.Code);
+                        Assert.Equal(PersistenceStatus.Persisted, dr.Status);
+                        Assert.Equal(topic.Name, dr.Topic);
+                        Assert.Equal(expectedPartition, (int)dr.Partition);
+                        Assert.True(dr.Offset >= 0);
+                        Assert.Equal(TimestampType.CreateTime, dr.Message.Timestamp.Type);
+                        Assert.True(Math.Abs((DateTime.UtcNow - dr.Message.Timestamp.UtcDateTime).TotalMinutes) < 1.0);
+                    };
+
+                    ProducerBuilder<string, string> producerBuilder = null;
+                    switch (j)
+                    {
+                        case 0:
+                            // Topic level custom partitioner.
+                            producerBuilder = new ProducerBuilder<string, string>(producerConfig);
+                            producerBuilder.SetPartitioner(topic.Name, (string topicName, int partitionCount, ReadOnlySpan<byte> keyData, bool keyIsNull) =>
+                            {
+                                Assert.Equal(topic.Name, topicName);
+                                var keyString = System.Text.UTF8Encoding.UTF8.GetString(keyData.ToArray());
+                                return int.Parse(keyString.Split(" ").Last()) % partitionCount;
+                            });
+                            break;
+                        case 1:
+                            // Default custom partitioner
+                            producerBuilder = new ProducerBuilder<string, string>(producerConfig);
+                            producerBuilder.SetDefaultPartitioner((string topicName, int partitionCount, ReadOnlySpan<byte> keyData, bool keyIsNull) =>
+                            {
+                                Assert.Equal(topic.Name, topicName);
+                                var keyString = System.Text.UTF8Encoding.UTF8.GetString(keyData.ToArray());
+                                return int.Parse(keyString.Split(" ").Last()) % partitionCount;
+                            });
+                            break;
+                        case 2:
+                            // Default custom partitioner in case where default topic config is present due to topic level config in top-level config.
+                            var producerConfig2 = new ProducerConfig
+                            {
+                                BootstrapServers = bootstrapServers
+                            };
+                            producerConfig2.Set("queuing.strategy", "lifo");
```

Comment:
This is deprecated and will be removed eventually.

Reply:
ok. the reason this got here is the test wasn't working and i was trying something obscure. changed.